### PR TITLE
CLOUDP-108867: Create one private endpoint for GCP Atlas

### PR DESCRIPTION
**Pull request by blva**
## Proposed changes

_Jira ticket:_ [CLOUDP-108867](https://jira.mongodb.org/browse/CLOUDP-108867)

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](e2e/E2E-TESTS.md) (if a new command or e2e test has been added)
- [x] I have run `make fmt` and formatted my code

## Further comments

```
➜  mongocli git:(master) ✗  mongocli atlas privateEndpoints gcp create --region ASIA_SOUTHEAST_2 --projectId 61dc598fd1f07c7b37ba425d
GCP private endpoint '61e582fd9d99eb1323b73098' created.
```

### CODE REVIEW

File: e2e/atlas/private_endpoint_test.go [hunk 1]
```diff
@@ -315,12 +315,71 @@ func TestPrivateEndpointsAzure(t *testing.T) {
...
+	"ASIA_SOUTHEAST_2",
+	"WESTERN_EUROPE",
+	"EUROPE_NORTH_1",
+	"EUROPE_WEST_2",
+	"EUROPE_WEST_3",
+	"EUROPE_WEST_4",
+	"EUROPE_WEST_6",
+	"EUROPE_CENTRAL_2",
+}
+
```
Comment (andreaangiolillo) [inline e2e/atlas/private_endpoint_test.go:348]: I think we agreed to add the e2e test only once you have added the delete command.
Keep in mind that not all the GPC regions are available on cloud-dev, you should test them on the UI before adding the region here.

File: internal/cli/atlas/privateendpoints/gcp/create.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+// mongocli atlas privateEndpoints gcp create --region region [--projectId projectId].
+func CreateBuilder() *cobra.Command {
+	opts := &CreateOpts{}
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: "Create a new GCP private endpoint for your project.",
+		Args:  require.NoArgs,
+		Example: `  This example uses the profile named "myprofile" for accessing Atlas.
+  $ mongocli atlas privateEndpoints gcp create --region CENTRAL_US`,
```
Comment (andreaangiolillo) [inline internal/cli/atlas/privateendpoints/gcp/create.go:74]: you are not using `myprofile` in the example. You should add `-P myprofile`

File: internal/cli/atlas/privateendpoints/gcp/create.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+				opts.ValidateProjectID,
+				opts.initStore(cmd.Context()),
+				opts.InitOutput(cmd.OutOrStdout(), createTemplate),
+			)
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return opts.Run()
+		},
+	}
+	cmd.Flags().StringVar(&opts.region, flag.Region, "", usage.PrivateEndpointRegion)
```
Comment (andreaangiolillo) [inline internal/cli/atlas/privateendpoints/gcp/create.go:86]:
```suggestion
	cmd.Flags().StringVar(&opts.region, flag.Region, "", usage.Region)
```

**Review by andreaangiolillo — COMMENTED** [review]
Just a few comments, nothing major. Great work!! 🥇

File: e2e/atlas/private_endpoint_test.go (hunk 1 above)
Comment (blva) [reply e2e/atlas/private_endpoint_test.go:348]: nice to know thanks for the heads up. just tested and will be removing 2 regions here.

File: internal/cli/atlas/privateendpoints/gcp/create.go (hunk 2 above)
Comment (blva) [reply internal/cli/atlas/privateendpoints/gcp/create.go:74]: oh ok, I guess I'll remove the whole line then as I got this from some previous docs. I'll keep the same pattern as the one in the list command if that makes sense?

**Review by andreaangiolillo — APPROVED** [review]
LGTM 💯

File: docs/command/mongocli-atlas-privateEndpoints-gcp-create.txt [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+     - false
+     - Output format. Valid values are json, json-path, go-template, or go-template-file.
+   * - --projectId
+     - string
+     - false
+     - Project ID to use. Overrides the settings in the configuration file or environment variable.
+   * - --region
+     - string
+     - true
+     - Physical location of your MongoDB cluster. For a complete list of supported AWS regions, see: https://docs.atlas.mongodb.com/reference/amazon-aws/#amazon-aws. For a complete list of supported Azure regions, see: https://docs.atlas.mongodb.com/reference/microsoft-azure/#microsoft-azure. For a complete list of supported GCP regions, see: https://docs.atlas.mongodb.com/reference/google-gcp/#google-gcp.
```
Comment (gssbzn) [inline docs/command/mongocli-atlas-privateEndpoints-gcp-create.txt:50]: [q] is this description correct? this is the description for cluster regions, just making sure you can use the same

Comment (andreaangiolillo) [reply docs/command/mongocli-atlas-privateEndpoints-gcp-create.txt:50]: yes, you are right. My fault here as Bianca added correctly `PrivateEndpointRegion` before my review. Not seeing the flag matching the usage tricks me

Comment (blva) [reply docs/command/mongocli-atlas-privateEndpoints-gcp-create.txt:50]: thanks! I'll move back to PrivateEndpointRegion

**Review by andreaangiolillo — APPROVED** [review]

**Pull request merged**